Chore: add ccm gas headsup (#455)
chainflip-io/chainflip-docs: pages/swapping/integrations/advanced/cross-chain-messaging/evm-ccm.mdx

## Gas budget estimation

> The user has to provide a gas budget for the destination chain that must cover the entire transaction on the destination chain, that is both the Chainflip signature verification and the execution of the user's logic. 

Reviewer: ```suggestion
The user has to provide a gas budget that must cover the entire transaction on the destination chain, that is both the Chainflip signature verification and the execution of the user's logic. 
```

> Currently that gas budget is provided in the source chain's native asset and is swapped by Chainflip into the destination chain's native asset to pay for gas. That gas budget is taken from the deposited amount.

Reviewer: ```suggestion
This gas budget is provided in the source chain's native asset and taken from the deposited amount. This amount is then swapped by Chainflip into the destination chain's native asset to pay for gas.
```

<Callout type="warning">

> To make the use of CCM easier and decrease the price and time risk, this mechanism will be upgraded so the gas budget will represent the amount of gas that the user wants to have on the destination chain. At the time of egress, Chainflip will take the minimum amount of the destination asset needed to cover for that gas. This makes estimations simpler, more reliable and cheaper for the users.

Reviewer: ```suggestion
**Upcoming upgrade on 1.8 release**: The gas budget will represent the amount on the destination chain instead. At the time of egress, Chainflip will take the minimum amount of the destination asset needed to cover the gas. This makes estimation simpler, cheaper and more reliable.
```

> Further detailed documentation will be provided when the feature included in the next release (1.8).

Reviewer: ```suggestion
Further detailed documentation will be provided in anticipation of the 1.8 release.
```
